indico/MaKaC/conference.py

```diff
@@ -3607,91 +3593,35 @@ def canKeyModify(self):
-                if sendEmail and isinstance(prin, ConferenceChair):
-                    notif = pendingQueues._PendingConfManagerNotification( [prin] )
-                    mail.GenericMailer.sendAndLog(notif, self.getConference(), 'Event')
-            #The user is registered in Indico and is activated as well
-            elif len(results) == 1 and results[0] is not None and results[0].isActivated():
-                self.__ac.grantModification(results[0])
-                results[0].linkTo(self, "manager")
-        else:
-            self.__ac.grantModification( prin )
-            if isinstance(prin, AvatarUserWrapper):
-                prin.linkTo(self, "manager")
-
-    def revokeModification( self, prin ):
-        self.__ac.revokeModification( prin )
-        if isinstance(prin, AvatarUserWrapper):
-            prin.unlinkTo(self, "manager")
-
-    @unify_user_args(legacy=True)
-    def canUserModify( self, av ):
-        if av == None:
-            return False
-        if ( av == self.getCreator()) or self.getAccessController().canModify( av ):
-            return True
-        for owner in self.getOwnerList():
-            if owner.canUserModify( av ):
-                return True
-        return False
+    @unify_user_args
+    def canUserModify(self, user, role=None):
+        return self.as_event.can_manage(user, role=role)
 
-    @unify_user_args(legacy=True)
-    def canModify(self, aw_or_user):
+    def canModify(self, aw_or_user, role=None):
         """Tells whether an access wrapper is allowed to modify the current
             conference: only if the user is granted to modify the conference and
             he is accessing from an IP address which is not restricted.
         """
         if hasattr(aw_or_user, 'getUser'):
             aw_or_user = aw_or_user.getUser()
-        return self.canUserModify(aw_or_user) or self.canKeyModify()
+        if isinstance(aw_or_user, AvatarUserWrapper):
+            aw_or_user = aw_or_user.user
+        return self.as_event.can_manage(aw_or_user, role=role, allow_key=True)
 
-    def getManagerList( self ):
-        return self.__ac.getModifierList()
-
-    def addToRegistrars(self, av):
-        self.getRegistrarList().append(av)
-        self.notifyModification()
-        if isinstance(av, AvatarUserWrapper):
-            av.linkTo(self, "registrar")
-
-    def removeFromRegistrars(self, av):
-        self.getRegistrarList().remove(av)
-        self.notifyModification()
-        if isinstance(av, AvatarUserWrapper):
-            av.unlinkTo(self, "registrar")
-
-    def isRegistrar(self, av):
-        if av is None:
-            return False
-        try:
-            return any(principal.containsUser(av) for principal in self.getRegistrarList())
-        except AttributeError:
-            return False
+    def getManagerList(self):
+        managers = sorted([x.principal for x in self.as_event.acl_entries if x.has_management_role()],
```

Comment:
I don't like that you'd have to specify a different argument. I think the first version is a lesser evil :confused: